flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/WindowAssigner.java

```diff
@@ -55,6 +55,12 @@ public abstract Collection<W> assignWindows(
             T element, long timestamp, WindowAssignerContext context);
 
     /** Returns the default trigger associated with this {@code WindowAssigner}. */
+    public Trigger<T, W> getDefaultTrigger() {
+        return getDefaultTrigger(new StreamExecutionEnvironment());
+    }
+
+    /** Returns the default trigger associated with this {@code WindowAssigner}. */
+    @Deprecated
```

Comment:
We should also explain the migration path to users. Particularly, after migrating to new methods, we need to keep an empty old method (because it's abstract), until the deprecated method is removed.